src/index.ts

```diff
@@ -0,0 +1,60 @@
+import { Context, Middleware } from '@curveball/core';
+import { BadRequest, NotFound } from '@curveball/http-errors';
+import fs from 'fs';
+import mime from 'mime-types';
+import path from 'path';
+
+export type Options = {
+  staticDir: string,
+}
+
+export default function serveFiles(options: Options): Middleware {
+  return async (ctx: Context, next) => {
+    serve(options.staticDir, ctx);
+    return next();
+
+  };
+}
+
+function serve(staticDir: string, ctx: Context) {
+  const pathPieces = ctx.request.path.split('/');
+
+  if (!staticDir.endsWith(pathPieces[1])) {
+    return;
+  }
+
+  const filePath = staticDir + '/' + pathPieces.slice(2).join('/');
```

Comment:
It would be nice, if we could map:

    http://localhost/foo/hello.jpg

to

    /home/hum/assets/hello.jpg

Would this allow that?

The ideal world would be to make this worth with the router middleware as well, basically:

    app.use(router('/static', static({
      staticDir: path.join(process.cwd(), 'assets')
    );

Reply:
Also, I think I can escape the jail by going to: `/assets/../../../../../etc/passwd`

Reply:
Yeah, this isn't going to work. This is one of the things I need to change. It won't work with nested folders.

What I'm trying to do is have the user config a static folder and then only requests to that static folder and it's children in the hierarchy can be returned. 

So for your example, that would *not* work. The static folder would be `assets` and the file requested would be `/foo/hello.jpg`. Since `foo` != `assets`, it would return a 404 (or be handled by a different route if it matches).

If my assets folder is `/home/hum/assets/`, things that should work:
- `/assets/hello.jpg`
- `/assets/deeper/hello.jpg`
-` /assets/deeper/../hello.jpg`

Things that should not work:
- `/assets/`
- `/assets/deeper`
- `/foo/hello.jpg`
- `/assets/../../../../../etc/passwd`